tests/run-make/const_fn_mir/rmake.rs

```diff
@@ -0,0 +1,8 @@
+// The `needs-unwind -Cpanic=abort` gives a different MIR output.
+
+use run_make_support::{cwd, diff, rustc};
+
+fn main() {
+    rustc().input("main.rs").emit("mir").output("dump.mir").run();
+    diff().expected_file("dump.mir").actual_file(cwd().join("dump.mir")).run();
```

Comment:
Problem: so we updated how rmake.rs tests are run, in that they are now setup like

```
build/<target>/.../<test-name>/
    rmake.exe
    rmake_out/
```

the rmake.rs test has $CWD set to `rmake_out/`, and input files from test sources are copied over to `rmake_out`. So when this rmake.rs test is about to run, the `rmake_out` ($CWD of the test) has

```
rmake_out/
    dump.mir
    main.rs
```

and `rustc().input("main.rs").emit("mir").output("dump.mir").run()` would override the reference `dump.mir`. Instead, I think we would need to:

```rs
let input_file = cwd().join("main.rs");
let reference_mir = cwd().join("dump.mir");
run_in_tmpdir(|| {
    // here the `$CWD` is adjusted to be some temporary directory `../temporary_directory` that is beside `rmake_out/`
    rustc().input(input_file).emit("mir").output("dump.mir").run();
    // here the generated `dump.mir` is `temporary_directory/dump.mir`, which does
    // not overwrite the reference mir `rmake_out/dump.mir`
    diff().expected_file(reference_mir).actual_file("dump.mir").run();
});
```

Reply:
It would probably be much simpler to just do `output("dump-actual.mir")` (or rename the reference file) to avoid the name conflict, than to run the whole thing in another temporary directory.

Reply:
Right, that is probably way easier.